### PR TITLE
Update mapviewer.py to Python 3

### PR DESCRIPTION
**Pull request by schneiderfelipe**
This also complies with PEP8 and avoids warnings (such as Gtk deprecations).

### CODE REVIEW

File: examples/mapviewer.py [hunk 1]
```diff
@@ -17,28 +17,33 @@
...
+gi.require_version('OsmGpsMap', '1.0')
+from gi.repository import Gtk  # noqa
+from gi.repository import OsmGpsMap as osmgpsmap  # noqa
 
-from gi.repository import OsmGpsMap as osmgpsmap
-print "using library: %s (version %s)" % (osmgpsmap.__file__, osmgpsmap._version)
+from gi.repository import Gdk  # noqa
+from gi.repository import GdkPixbuf  # noqa
+from gi.repository import GObject  # noqa
+from gi.repository import GLib  # noqa
```
Comment (bittner) [inline examples/mapviewer.py]: Is it important that you make all separate imports here? Would a combined import like that be a (style) issue?

```python
import gi

gi.require_version('Gdk', '3.0')
gi.require_version('Gtk', '3.0')
gi.require_version('OsmGpsMap', '1.0')

from gi.repository import (
    Gdk, GdkPixbuf, GLib, GObject, Gtk, OsmGpsMap as osmgpsmap
)  # noqa
```

File: examples/mapviewer.py [hunk 2]
```diff
@@ -17,28 +17,33 @@
...
+
+print("using library: {} (version {})".format(osmgpsmap.__file__,
+                                              osmgpsmap._version))
 
 assert osmgpsmap._version == "1.0"
```
Comment (bittner) [inline examples/mapviewer.py]: I know this is not part of your changeset, but ...

Do we need this line actually? It won't / shouldn't be relevant for production code, hence why keep it? Only confuses newcomers.

File: examples/mapviewer.py [hunk 3]
```diff
@@ -189,29 +204,29 @@ def load_map_clicked(self, button):
         format = self.image_format_entry.get_text()
         if uri and format:
             if self.osm:
-                #remove old map
+                # remove old map
                 self.vbox.remove(self.osm)
```
Comment (bittner) [inline examples/mapviewer.py]: The sole fact that there is a comment indicates that we need to wrap the whole management of `self.vbox` into an object or helper functions. Such a trivial action like removing an object should really [be self-explanatory and not need a comment](https://softwareengineering.stackexchange.com/a/51313/23443).

File: examples/mapviewer.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
```
Comment (bittner) [inline examples/mapviewer.py]: Should maybe be `#!/usr/bin/env python3` to make it work intuitively (e.g. the current implementation will break in a virtualenv).

File: examples/mapviewer.py (hunk 1 above)
Comment (schneiderfelipe) [reply examples/mapviewer.py]: Done.

File: examples/mapviewer.py (hunk 2 above)
Comment (schneiderfelipe) [reply examples/mapviewer.py]: Do you mean the assert line? If yes, I agree. In fact, the `gi.require_version('OsmGpsMap', '1.0')` above makes this line a little silly :)

Done.

File: examples/mapviewer.py [hunk 5]
```diff
@@ -17,28 +17,33 @@
...
-from math import pi
-from gi.repository import Gtk
-from gi.repository import Gdk
-from gi.repository import GdkPixbuf
-from gi.repository import GObject
+import gi
 
-GObject.threads_init()
-Gdk.threads_init()
+gi.require_version('Gtk', '3.0')
```
[the remark this answers is not in the document]
Comment (schneiderfelipe) [reply examples/mapviewer.py]: You're right. Done.

File: examples/mapviewer.py (hunk 4 above)
Comment (schneiderfelipe) [reply examples/mapviewer.py]: Done.

File: examples/mapviewer.py (hunk 3 above)
Comment (schneiderfelipe) [reply examples/mapviewer.py]: I didn't quite understand this. I removed the comment, as I believe the code to be clear enough.

**Comment by schneiderfelipe** [comment]
Thanks for your comments. I tried to address everything.
By the way, if too many commits is an issue, I can rebase everything into a single one, just let me know.

File: examples/mapviewer.py [hunk 6]
```diff
@@ -189,29 +208,26 @@ def load_map_clicked(self, button):
...
-                self.osm = osmgpsmap.Map(
-                    repo_uri=uri,
-                    image_format=format
-                )
-            except Exception, e:
-                print "ERROR:", e
-                self.osm = osm.Map()
+                self.osm = osmgpsmap.Map(repo_uri=uri, image_format=format)
+            except Exception as e:
+                print("ERROR:", e)
```
Comment (bittner) [inline examples/mapviewer.py]: Just for the record: Writing the `print` statements like this isn't backwards-compatible with Python 2.

Python 2:
```
>>> print("foo", "bar")
('foo', 'bar')
```
Python 3:
```
>>> print("foo", "bar")
foo bar
```
It writes out with Python 3, though, what was written out with the old code on Python 2:
```
>>> print "foo", "bar"
foo bar
```
Let's agree that Python 2 is meant to be dead for this example. :checkered_flag:

Comment (bittner) [reply examples/mapviewer.py]: ... the f-strings below require Python >= 3.6 anyway. True.

**Comment by bittner** [comment]
> By the way, if too many commits is an issue, I can rebase everything into a single one, just let me know.

I'm not a maintainer of this project but, yes, I'd agree it would make sense to squash all your changes into a single commit.

**Comment by bittner** [comment]
@nzjrs, anything missing for this PR to be merged?

**Comment by schneiderfelipe** [comment]
> I'm not a maintainer of this project but, yes, I'd agree it would make sense to squash all your changes into a single commit.

@bittner @nzjrs I squashed all changes to a single commit as suggested. Please let me know if there's anything else to be done.

**Comment by bittner** [comment]
Very cool! -- Now we just need this merged. @nzjrs?

**Comment by johnny-bit** [comment]
Works for me ;)

**Pull request merged**